tests/test_registration.py

```
@@ -0,0 +1,532 @@
+    assert default_handler_registry.get_query_handler(SubSampleQuery) is None
+
+    # If you want to handle subclasses, they must be registered explicitly
+    @command_handler(SubSampleCommand)
+    class MySubSampleCommandHandler(CommandHandler[SubSampleCommand]):
+        def __call__(self, request: SubSampleCommand) -> None: pass
+    
+    assert default_handler_registry.get_command_handler(SubSampleCommand) is MySubSampleCommandHandler
+
+# Test that type checks in decorators and registry methods are actually working
+def test_type_check_enforcement_in_decorators():
+    class NotACommand: pass
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)